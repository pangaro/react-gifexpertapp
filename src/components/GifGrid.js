import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category } ) => {

    const { data:images, loading } = useFetchGifs( category );

    //console.log(state);
/*
    const [images, setimages] = useState([]);

    useEffect( () => {
        getGifs( category )
            .then( setimages );
            //.then( ( imgs ) => setimages( imgs ) );
    }, [ category ]);
*/

    return (
        <>
            <h3 className="animate__animated animate__swing__delay-2s"> { category } </h3>

            { loading && <p>Loading...</p> }
            <div className="card-grid">
                {
                     images.map( ( img ) => (
                    <GifGridItem 
                        key={ img.id }
                        { ...img } /> 
                    ))
                }
            </div>
        </>
    )
}
