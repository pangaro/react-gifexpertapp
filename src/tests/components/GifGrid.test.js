import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en <GifGrid />', () => {
    
    const category = 'goku';

    test('debe mostrar correctamente <GifGrid /> ', () => {
   
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
    
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg', 
            title: 'Cualquier cosa'
        }];


        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
    
});
