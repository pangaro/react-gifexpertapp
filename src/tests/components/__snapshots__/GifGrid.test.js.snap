// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe mostrar correctamente <GifGrid />  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing__delay-2s"
  >
     
    goku
     
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing__delay-2s"
  >
     
    goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
